### PR TITLE
Remove the redundant copy of join key column

### PR DESCRIPTION
**Pull request by gengliqi**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflash/issues/9057

Problem Summary:
1. fix https://github.com/pingcap/tiflash/issues/9057
2. this PR also removes the redundant column names in `output_columns_names_set_for_other_condition_after_finalize`.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/TestUtils/FunctionTestUtils.cpp [hunk 1]
```diff
@@ -243,10 +243,13 @@ ::testing::AssertionResult columnsEqual(
...
-            expect_col.column->getName(),
-            actual_col.column->getName(),
-            fmt::format("Column {} name mismatch", i));
+        if (expect_col.column->getName() != actual_col.column->getName())
+        {
+            ASSERT_EQUAL(
+                expect_col.column->getName(),
+                actual_col.column->getName(),
+                fmt::format("Column {} name mismatch", i));
+        }
```
Comment (gengliqi) [inline dbms/src/TestUtils/FunctionTestUtils.cpp:252]:
```suggestion
        ASSERT_EQUAL(
            expect_col.column->getName(),
            actual_col.column->getName(),
            fmt::format("Column {} name mismatch", i));
```

File: dbms/src/TestUtils/FunctionTestUtils.cpp [hunk 2]
```diff
@@ -243,10 +243,13 @@ ::testing::AssertionResult columnsEqual(
     {
         auto const & expect_col = expected[i];
         auto const & actual_col = actual[i];
-        ASSERT_EQUAL(
-            expect_col.column->getName(),
-            actual_col.column->getName(),
-            fmt::format("Column {} name mismatch", i));
+        if (expect_col.column->getName() != actual_col.column->getName())
```
[the remark this answers is not in the document]
Comment (gengliqi) [reply dbms/src/TestUtils/FunctionTestUtils.cpp]: For debug, forget to change it back.

**Review by windtalker — APPROVED** [review]
LGTM

**Review by yibin87 — APPROVED** [review]
LGTM

**Pull request merged**